### PR TITLE
Refactor WoodenA and WoodenB paint setup

### PR DESCRIPTION
**Pull request by Gymnasiast**
Most of the differences were stylistic. I compared both with `meld` and updated comments, variable names, if/else order and stuff like that to make them as similar as possible. The actual differences turned out to be very small. I also put some common code into static functions to reduce their size further.

More refactoring is certainly possible, but that would require a bit more digging into why the differences are there in the first place.

It is recommend to look at the commits one by one.

### CODE REVIEW

File: src/openrct2/paint/Supports.cpp [hunk 1]
```diff
@@ -634,7 +644,7 @@ bool WoodenBSupportsPaintSetup(
         imageTemplate = ImageId().WithTransparency(FilterPaletteID::PaletteDarken1);
     }
 
-    uint16_t baseHeight = Ceil2(session.Support.height, 16);
+    auto baseHeight = Floor2(session.Support.height + 15, 16);
```
Comment (tupaschoal) [inline src/openrct2/paint/Supports.cpp]: Most of the changes on commit one were stylistic, but this one was an actual change, can you elaborate on it?

File: src/openrct2/paint/Supports.cpp [hunk 2]
```diff
@@ -473,7 +551,7 @@ bool WoodenASupportsPaintSetup(
         imageTemplate = ImageId().WithTransparency(FilterPaletteID::PaletteDarken1);
     }
 
-    auto baseHeight = Floor2(session.Support.height + 15, 16);
+    uint16_t baseHeight = Floor2(session.Support.height + 15, 16);
```
Comment (tupaschoal) [inline src/openrct2/paint/Supports.cpp:554]: Seems like this changed is tied to what happened on commit 1 and should be moved there.

**Review by tupaschoal — COMMENTED** [review]
Looks like a great improvement in code organization, I have just one question

File: src/openrct2/paint/Supports.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/paint/Supports.cpp]: When you run through all values, it actually comes out as exactly the same value:

For `Floor2(session.Support.height + 15, 16);`:
orig. height|parameter 1|result
---|---|---
0|15|0
1|16|16
2|17|16
16|31|16
17|32|32
18|33|32

For `Ceil2(session.Support.height, 16);`
orig. height|result
---|---
0|0
1|16
2|16
16|16
17|32
18|32

**Review by tupaschoal — APPROVED** [review]

File: src/openrct2/paint/Supports.cpp (hunk 1 above)
Comment (spacek531) [reply src/openrct2/paint/Supports.cpp]: I am curious what the benefit of this is

Comment (Gymnasiast) [reply src/openrct2/paint/Supports.cpp]: We had two functions that were 99% the same, but using different names, comments, etc. I first eliminated the difference, and then moved some parts that were the same into functions of their own. The result is clearer code and less duplicate code. That’s the benefit.

**Pull request merged**